bundles/org.openhab.binding.daikin/src/main/java/org/openhab/binding/daikin/internal/DaikinWebTargets.java

```diff
@@ -49,13 +61,29 @@
     private String getAirbaseZoneInfoUri;
     private String setAirbaseZoneInfoUri;
 
+    private String uuid;
+    private static HttpClient httpClient;
+
     private Logger logger = LoggerFactory.getLogger(DaikinWebTargets.class);
 
-    public DaikinWebTargets(String ipAddress) {
-        String baseUri = "http://" + ipAddress + "/";
+    public DaikinWebTargets(String host, Boolean secure, String uuid) {
+        if (httpClient == null) {
+            httpClient = new HttpClient(new SslContextFactory(true));
```

Comment:
You should be creating client instances from the osgi `org.eclipse.smarthome.io.net.http.HttpClientFactory` service.

Reply:
I tried this. It didn't work because we need "trust all" from SslContextFactory. We're dealing with name mismatch - the adapter is accessed using https://192.168.1.x (ip address) and possibly self signed certificate. The Common Name (CN) of the certificate is something like "0ADF1002341228818192.srv", so clearly no way to get this name to match without creating a static DNS entry on the user's network. I have even tried `httpClient.getSslContextFactory().setTrustAll(true)` but that didn't seem to work. So this method of creating a custom HttpClient is the only way that seems to work.

Reply:
Ok, since you had a good reason to create the client instance yourself I'll allow it.